authentication/managing_cloud_provider_credentials/cco-mode-sts.adoc

```diff
@@ -0,0 +1,734 @@
+
+In manual mode with STS, the individual {product-title} cluster components use AWS Secure Token Service (STS) to assign components IAM roles that provide short-term, limited-privilege security credentials. These credentials are associated with IAM roles that are specific to each component that makes AWS API calls.
+
+Requests for new and refreshed credentials are automated by using an appropriately configured AWS IAM OpenID Connect (OIDC) identity provider, combined with AWS IAM roles. {product-title} signs service account tokens that are trusted by AWS IAM, and can be projected into a pod and used for authentication. Tokens are refreshed after one hour.
+
+.STS authentication flow
+image::142_OpenShift_credentials_STS_0221.svg[Detailed authentication flow between AWS and the cluster when using AWS STS]
+
+Using manual mode with STS changes the content of the AWS credentials that are provided to individual OpenShift components.
+
+.AWS secret format using long-lived credentials
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  namespace: <target-namespace> <1>
+  name: <target-secret-name> <2>
+data:
+  aws_access_key_id: <base64-encoded-access-key-id>
+  aws_secret_access_key: <base64-encoded-secret-access-key>
+----
+<1> The namespace for the component.
+<2> The secret name for the component.
+
+.AWS secret format with STS
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  namespace: <target-namespace> <1>
+  name: <target-secret-name> <2>
+data:
+  role_name: <operator-role-name> <3>
+  web_identity_token_file: <path-to-token> <4>
+----
+<1> The namespace for the component.
+<2> The secret name for the component.
+<3> The IAM role for the component.
+<4> The path to the service account token inside the pod. By convention, this is `/var/run/secrets/openshift/serviceaccount/token` for {product-title} components.
+
+[id="sts-mode-installing"]
+== Installing an {product-title} cluster configured for manual mode with STS
+
+In {product-title} version 4.7, you can install a cluster that is configured to use the CCO in manual mode with STS by using one of two methods:
+
+* xref:#sts-mode-installing-preflight-config[Configuring a cluster with the `sts-preflight` tool]
+* xref:#sts-mode-installing-manual-config[Configuring a cluster manually]
+
+After installing with either of these methods, the procedure for xref:#sts-mode-installing-verifying[verifying the installation] is the same.
+
+[id="sts-mode-installing-preflight-config"]
+=== Configuring a cluster with the sts-preflight tool
+
+When xref:#sts-mode-installing[installing] an {product-title} cluster that is configured to use the CCO in manual mode with STS, you can use the `sts-preflight` tool to create the required AWS resources.
+
+This tool, which is hosted in an external repository, is provided to help configure a cluster that uses STS mode while the feature is a link:https://access.redhat.com/support/offerings/techpreview[Technology Preview].
```

Comment:
Do we need _is provided to_? 
_This tool... helps configure a cluster..._

_while the feature is a_ This sounds like the tool will go away at GA. Is this the case?

Reply:
The tool is available as long as github stays up and running, or as long as the developer doesn't archive/delete the repo. But long-term, we are developing a different tool to handle what sts-preflight does.